dateparser/date_parser.py

```diff
@@ -32,10 +33,103 @@ def __add__(self, other):
 parser.relativedelta.relativedelta = new_relativedelta
 
 
+class new_parser(parser.parser):
+    """
+    Implements an alternate parse method which supports preference to dates in future and past.
+    For more see issue #36
+    """
+
+    def parse(self, timestr, default=None, ignoretz=False, **kwargs):
+        # timestr needs to be a buffer as required by _parse
+        if isinstance(timestr, binary_type):
+            timestr = timestr.decode()
+
+        # Parse timestr
+        res = self._parse(timestr, **kwargs)
+
+        if res is None:
+            raise ValueError("unknown string format")
+
+        # Fill in missing date
+        new_date, given_fields = new_parser._populate(res, default)
+
+        # Correct date to if prefer in future of past
+        new_date = new_parser._correct(new_date, given_fields, default)
+
+        # Clean hour and minutes, etc in case not defined
+        if not res.hour and not res.minute and not res.second and not res.microsecond:
+            new_date = new_date.replace(hour=0, minute=0, second=0, microsecond=0)
+
+        return new_date
+
+    @staticmethod
+    def _populate(res, default):
+        given_fields = []  # fields that cannot be overriden
+        new_date = default
+
+        # Populate all fields
+        repl = {}
+        for field in ['year', 'month', 'day', 'hour', 'minute', 'second', 'microsecond']:
+            value = getattr(res, field)
+            if value is not None:
+                repl[field] = value
+        new_date = new_date.replace(**repl)
+
+        # Fix weekday
+        if res.weekday and not res.day:
+            new_date = new_date + new_relativedelta(weekday=res.weekday)
+
+        return new_date, [key + 's' for key in repl.keys()]
+
+    @staticmethod
+    def _correct(date, given_fields, default):
+        if settings.PREFER_DATES_FROM == 'current_period':
+            return date
+
+        for field in ['microseconds', 'seconds', 'minutes', 'hours', 'days',
+                      'weeks', 'months', 'years']:
+            # Can't override a given field
+            if field in given_fields:
+                continue
+
+            # Try if applying the delta for this field corrects the problem
+            delta = relativedelta(**{field: 1})
+            new_date = new_parser._correct_for_future(date, delta, default) or new_parser._correct_for_past(date, delta, default)
+            if new_date:
+                date = new_date
+                break
+
+        return date
+
+    @staticmethod
+    def _correct_for_future(date, delta, default):
+        if settings.PREFER_DATES_FROM != 'future':
+            return None
+
+        if date < default < date + delta:
+            date += delta
+            return date
+
+        return None
+
+    @staticmethod
+    def _correct_for_past(date, delta, default):
+        if settings.PREFER_DATES_FROM != 'past':
+            return None
```

Comment:
Returning unchanged date as i suggested here instead of `None` would go better with method name `correct_*` and would allow not to check the result of this function in base `_correct` method but simply use several `date = correc_*(date)`